ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+from .models import OrderedProduct
+from client.models import Client
+from supplier_product.models import SupplierProduct
+from delivery_location.models import DeliveryLocation
+from product.models import Product
+from supplier.models import Supplier
+from django.shortcuts import redirect
+from django.shortcuts import render
+from django.contrib.auth.decorators import user_passes_test
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.models import User
+
+
+def client_check(user):
+    try:
+        return Client.objects.get(client_account=user)
+    except Exception:
+        return False
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def change_quantity(request, order_id):
+    try:
+        quantity = int(request.GET['quantity'])
+        ordered_product = OrderedProduct.objects.get(id=int(order_id))
+    except Exception:
```

Comment:
narrow down the exception

Reply:
IN this case I want to catch ObjectDoesNotExist and illegal conversion to int.

Reply:
I know but in here if i gave wrong order_id that does not exist i will be redirected without knowing what happend

Reply:
You are not able to put wrong order_id beacuse the page show for each order a bottom to change quantity.
I did that only if you put wrong order_id manually in the URL. Client usually don't do this.

Reply:
> IN this case I want to catch ObjectDoesNotExist and illegal conversion to int.

```suggestion
    except ObjectDoesNotExist:
        # handle this
    except ValueError:
        # handle that
```